common/src/main/java/org/apache/comet/parquet/ColumnReader.java

```diff
@@ -172,28 +172,6 @@ public void close() {
 
   /** Returns a decoded {@link CometDecodedVector Comet vector}. */
   public CometDecodedVector loadVector() {
-    // Only re-use Comet vector iff:
```

Comment:
I don't know how much performance gain we got from reusing `CometVector` here. Note that it doesn't skip loading data into buffer in native side. It just reuses JVM `CometVector`.

This reuse conflicts safety check. Because in CometColumnarToRow, we close columnar vectors to remove buffer reference in JVM after accessing them. `loadVector` is called after that to load next columnar vector (i.e., `CometVector`). As it is closed, we cannot reuse it now.

Reply:
Even though the conditions for reuse are extremely well documented here, it is still confusing to keep track of memory while debugging. I seem to remember that there was some gain from the memory reuse, but if more recent benchmarks show no gain, should we remove this reuse (and simplify management of our buffer lifetimes)?

Reply:
This needs to be removed. As I explained above early, without removing it, we cannot close columnar vectors in `CometColumnarToRow` and it is required to pass safety check.